net/httpclient.go

```diff
@@ -21,14 +21,66 @@ const (
 	defaultRefreshInterval = 5 * time.Minute
 )
 
+// Lookuper is an indirection and used to map calls to
+// secrets.SecretsReader's GetSecret to the right parameter.
+type Lookuper interface {
```

Comment:
from the 'net' packages perspective, 'net.Lookuper' is not very descriptive. It could be rather 'SecretLookuper'

Reply:
I agree, changed